### ISSUE TITLE
test fails with "stream did not contain valid UTF-8" since 0.14.* on macOS

### ISSUE THREAD

**Issue by AlexanderOMara** [tree-sitter/tree-sitter#326]
I've been trying to update tree-sitter-cli past version 0.13.16, but the versions after do not seem to be working, at least on macOS. All I get is the error without any other explanation:

```
stream did not contain valid UTF-8
```

If I manually install an older version, it works again.

This happens with a very simple example grammar:

```js
module.exports = grammar({
	name: 'the_language_name',

	rules: {
		source_file: $ => repeat($._definition),

		_definition: $ => choice(
			$.function_definition
			// TODO: other kinds of definitions
		),

		function_definition: $ => seq(
			'func',
			$.identifier,
			$.parameter_list,
			$._type,
			$.block
		),

		parameter_list: $ => seq(
			'(',
			 // TODO: parameters
			')'
		),

		_type: $ => choice(
			$.primitive_type,
			$.array_type,
			$.pointer_type
		),

		primitive_type: $ => choice(
			'bool',
			'int'
		),

		array_type: $ => seq(
			'[',
			']',
			$._type
		),

		pointer_type: $ => seq(
			'*',
			$._type
		),

		block: $ => seq(
			'{',
			repeat($._statement),
			'}'
		),

		_statement: $ => choice(
			$.return_statement
			// TODO: other kinds of statements
		),

		return_statement: $ => seq(
			'return',
			$._expression,
			';'
		),

		_expression: $ => choice(
			$.identifier,
			$.number
			// TODO: other kinds of expressions
		),

		identifier: $ => /[a-z]+/,

		number: $ => /\d+/
	}
});
```

And test file:

```
==================
Return statements
==================

func x() int {
  return 1;
}

---

(source_file
  (function_definition
    (identifier)
    (parameter_list)
    (primitive_type)
    (block
      (return_statement (number)))))
```

I suspect this issue may be macOS related, since it turns out it still working on Travis CI using Linux.

Any idea what the breaking change may be?

Tested with:
Node v11.15.0
macOS 10.14.4

**Comment by maxbrunsfeld** [comment]
Hmm, yeah I need to add some context information to all of the error messages. Are all of your `corpus` files valid UTF8?

**Comment by AlexanderOMara** [comment]
Yep, that was the first thing I checked.

```
$ file corpus/*
corpus/return-statements.txt: ASCII text
$ xxd corpus/return-statements.txt
00000000: 3d3d 3d3d 3d3d 3d3d 3d3d 3d3d 3d3d 3d3d  ================
00000010: 3d3d 0a52 6574 7572 6e20 7374 6174 656d  ==.Return statem
00000020: 656e 7473 0a3d 3d3d 3d3d 3d3d 3d3d 3d3d  ents.===========
00000030: 3d3d 3d3d 3d3d 3d0a 0a66 756e 6320 7828  =======..func x(
00000040: 2920 696e 7420 7b0a 2020 7265 7475 726e  ) int {.  return
00000050: 2031 3b0a 7d0a 0a2d 2d2d 0a0a 2873 6f75   1;.}..---..(sou
00000060: 7263 655f 6669 6c65 0a20 2028 6675 6e63  rce_file.  (func
00000070: 7469 6f6e 5f64 6566 696e 6974 696f 6e0a  tion_definition.
00000080: 2020 2020 2869 6465 6e74 6966 6965 7229      (identifier)
00000090: 0a20 2020 2028 7061 7261 6d65 7465 725f  .    (parameter_
000000a0: 6c69 7374 290a 2020 2020 2870 7269 6d69  list).    (primi
000000b0: 7469 7665 5f74 7970 6529 0a20 2020 2028  tive_type).    (
000000c0: 626c 6f63 6b0a 2020 2020 2020 2872 6574  block.      (ret
000000d0: 7572 6e5f 7374 6174 656d 656e 7420 286e  urn_statement (n
000000e0: 756d 6265 7229 2929 2929 0a              umber))))).
```

**Comment by maxbrunsfeld** [comment]
Sorry for the poor experience there. I think that the new version of `tree-sitter-cli`, 0.15.2, will have better error messages, so hopefully we'll be better able to tell what file is causing the problem. It will be published to npm shortly.

**Comment by spencertipping** [comment]
I had the same problem caused by vim swapfiles: #369. I'm not sure if it's causing your problem, but it might be related.

**Comment by maxbrunsfeld** [comment]
I believe this was fixed by #430.
